avocado/core/requirements.py

```diff
@@ -0,0 +1,94 @@
+
+import argparse
+import os
+
+from . import data_dir
+from . import safeloader
+from ..utils.asset import Asset
+from ..utils.vmimage import get as get_vmimage
+from ..utils import data_structures
+
+
+class RequirementsResolver:
+
+    def __init__(self, references):
+        self.references = references
+        self.supported_requirements = {
+            'file': self._handle_assets,
+            'image': self._handle_vmimage,
+        }
+
+    @staticmethod
+    def _extract_requirements_from_reference(test):
```

Comment:
In line with my previous comment, this is, as you know, already done in `avocado.core.resolvers.PythonUnittestResolver()` and `avocado.core.resolvers.AvocadoInstrumentedResolver()`.  I feel this is the right place, because let's say, a `ExecTestResolver` may have a different way of setting requirements, with the following extra code:

```diff
diff --git a/avocado/plugins/resolvers.py b/avocado/plugins/resolvers.py
index c16084f5..3b864b15 100644
--- a/avocado/plugins/resolvers.py
+++ b/avocado/plugins/resolvers.py
@@ -44,9 +44,13 @@ class ExecTestResolver(Resolver):
         if criteria_check is not True:
             return criteria_check
 
+        requirements_file = os.path.join(reference, '.data', 'requirements.txt')
+        reqs = RequirementsResolver.load_requirements_from_file(requirements_file)
+
         return ReferenceResolution(reference,
                                    ReferenceResolutionResult.SUCCESS,
-                                   [Runnable('exec-test', reference)])
+                                   [Runnable('exec-test', reference,
+                                             requirements=reqs)])
 
 
 class PythonUnittestResolver(Resolver):
```

Reply:
Yep, I agree. The JSON representation from BP002 should cover this case (https://avocado-framework.readthedocs.io/en/latest/blueprints/BP002.html#requirements-representation-as-json-files)